platform/lib/platform/material.ex

```diff
@@ -385,6 +388,50 @@ defmodule Platform.Material do
     end
   end
 
+  def create_media_audited(%APIToken{} = api_token, attrs, opts) do
+    changeset =
+      %Media{}
+      |> Media.changeset(attrs, api_token)
+
+    cond do
+      !changeset.valid? ->
+        {:error, changeset}
+
+      true ->
+        Repo.transaction(fn ->
```

Comment:
I'm a bit hesitant to have the ~same logic repeated in two places — can we combine the "business logic" into a function that all versions of `create_media_audited` call?